tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/bulkhead/clientserver/Checker.java

```diff
@@ -82,41 +85,36 @@ public Checker(int sleepMillis, TestData td, int fails) {
             while ((now > max) && !td.getMaxSimultaneousWorkers().compareAndSet(max, now)) {
                 max = td.getMaxSimultaneousWorkers().get();
             }
+
             if (fails > 0) {
                 Thread.sleep(millis / 2);
                 fails--;
                 RuntimeErrorException e = new RuntimeErrorException(new Error("fake error for Retry Testing"));
-                Utils.log(e.toString());
-
+                log(e.toString());
                 // We will countDown the latch in the finally block
-
                 throw e;
             }
 
-            Utils.log("Task " + taskId + " sleeping for " + millis + " milliseconds. " + now
-                    + " workers inside Bulkhead from " + td.getInstances() + " instances " + BAR.substring(0, now));
-
-            Thread.sleep(millis);
+            log("Task " + taskId + " sleeping for " + millis + " milliseconds. " + now
+                + " workers inside Bulkhead from " + td.getInstances() + " instances " + BAR.substring(0, now));
 
-            Utils.log("Task " + taskId + " woke.");
+            await().atMost(millis, MILLISECONDS).untilAsserted(() -> assertTrue(fails < 0));
```

Comment:
This is weird. The `fails--` statement above is the only place where `fails` is modified. So at this point, either `fails < 0` already, or it isn't and it doesn't make sense to wait for it. Or am I missing something?

Reply:
It looks to me like the original code was sleeping to simulate some work, rather than sleeping to wait for something to happen.

If that's the case, then we'd just want to sleep rather than using awaitility.

If that's not the case, we should change the log line above which says we're going to sleep.

Reply:
Yea. We definitely shouldn't blindly replace all `Thread.sleep` -- if the code sleeps to simulate work, rewriting it to Awaitility is a bad idea. (It's possible, as shown here, but it will only obscure the meaning of the code and provide no benefit at all.)

Reply:
I put it back in the sleep part. done

Reply:
done